e2e-tests/src/test/staking.rs

```diff
@@ -195,3 +101,94 @@ pub fn staking_test(config: &Config) -> anyhow::Result<()> {
+    let stash_seed = "//Stash";
+    let stash = keypair_from_string(stash_seed);
+    let stash_account = AccountId::from(stash.public());
+    let (_, mut validator_accounts) = get_key_pairs();
+    let node = &config.node;
+    let sender = validator_accounts.remove(0);
+    let connection = create_connection(node).set_signer(sender);
+
+    send_change_members(
+        &connection,
+        convert_authorities_to_account_id(validator_accounts.clone()),
+    );
+
+    let current_session = get_current_session(&connection);
+    let _ = wait_for_session(&connection, current_session + 2)?;
+
+    // to cover tx fees as we need a bit more than VALIDATOR_STAKE
+    batch_endow_account_balances(&connection, &[stash.clone()], VALIDATOR_STAKE + TOKEN);
+    // to cover txs fees
+    batch_endow_account_balances(&connection, &[controller.clone()], TOKEN);
+
+    bond(node, VALIDATOR_STAKE, &stash, &controller);
+    let bonded_controller_account = bonded(&connection, &stash);
+    assert!(
+        bonded_controller_account.is_some(),
+        "Expected that stash account {} is bonded to some controller!",
+        &stash_account
+    );
+    let bonded_controller_account = bonded_controller_account.unwrap();
+    assert_eq!(
+        bonded_controller_account, controller_account,
+        "Expected that stash account {} is bonded to the controller account {}, got {} instead!",
+        &stash_account, &controller_account, &bonded_controller_account
+    );
+
+    let validator_keys = get_author_rotate_keys(&connection).unwrap().unwrap();
+    session_set_keys(node, &controller, validator_keys, XtStatus::InBlock);
+
+    // to be elected in next era instead of expected validator_account_id
+    validate(node, &controller, XtStatus::InBlock);
+
+    let ledger = ledger(&connection, &controller);
+    assert!(
+        ledger.is_some(),
+        "Expected controller {} configuration to be non empty",
+        controller_account
+    );
+    let ledger = ledger.unwrap();
+    assert_eq!(
+        ledger,
+        StakingLedger {
+            stash: stash_account.clone(),
+            total: VALIDATOR_STAKE,
+            active: VALIDATOR_STAKE,
+            unlocking: vec![],
+            // we don't need to compare claimed rewards as those are internals of staking pallet
+            claimed_rewards: ledger.claimed_rewards.clone()
+        }
+    );
+
+    validator_accounts.push(stash.clone());
+    send_change_members(
+        &connection,
+        convert_authorities_to_account_id(validator_accounts.clone()),
+    );
+    let current_session = get_current_session(&connection);
+    let _ = wait_for_session(&connection, current_session + 2)?;
+
+    let current_era = wait_for_full_era_completion(&connection)?;
+    info!(
+        "Era {} started, claiming rewards for era {}",
+        current_era,
+        current_era - 1
+    );
+
+    payout_stakers(node, stash, current_era - 1);
```

Comment:
Maybe we should check if there was some payout? Not exactly, just check if stash increased.

Reply:
Right, I was thinking the same. Let me do it.